### ISSUE TITLE
HostMemoryFull alert doesn't take into account available hugepages 

### ISSUE THREAD

**Issue by peppepetra** [canonical/grafana-agent-operator#41]
### Bug Description

In an environment where memory is preallocated as Hugepages, this is firing for all nodes although they have plenty of memory available as unused Hugepages.

### To Reproduce

NA

### Environment

NA

### Relevant log output

```shell
NA
```

**Comment by dstathis** [comment]
What we should do is have two alerts, one will trigger when hugepage memory is almost full and one that triggers when "non hugepage" memory is almost full.

**Comment by PietroPasotti** [comment]
We should document in the alert rule why it's done this way.

**Comment by lucabello** [comment]
@sed-i Did we agree on some approach for this?

**Comment by lucabello** [comment]
It's not super clear to me which type of alert we'd want to have:
1. We could change the alert to something like `(<used memory> - Hugetlb_bytes)/(<total memory> - Hugetlb_bytes) > 90%`, to alert when the non-Hugepages disk space is mostly full;
2. Alert when the space allocated via Hugepages is mostly full - which is not possible, as we have no metric for that.
3. Just disable the alert.


Hugetlb_bytes: This is the total amount of memory, consumed by huge pages of all sizes  https://www.rockdata.net/external/node_exporter-metrics/

**Comment by lucabello** [comment]
We will do option 1.

**Comment by marcusboden** [comment]
Option 1 sounds good. But can we have a hugetable usage alert as well? Something like `1- HugePages_Free / HugePages_Total > 10%`?

**Comment by lucabello** [comment]
@marcusboden Could you rephrase your alert in words? Do you mean:
1. The amount of available Hugepages (the number of pages, not the space) is too low (`HugePages_Free / `HugePages_Total < 10%`) (i.e. "there are less then 10% of hugepages available", in terms of number of pages);
2. The amount of available **space** that is preallocated with Hugepages is running out - as mentioned previously, we cannot compute this; most metrics refer to the number of pages, there are no metrics referring to the free/occupied space inside of Hugepages expressed in bytes.

AFAIU the first alert wouldn't be particularly useful, because you have no information about the size of Hugepages. If (1) is trying to achieve (2) but with different metrics, then we shouldn't add it. However, if (1) on its own is a legitimately useful alert, please do tell!
